package ZPDataControl;

message Node {
  required bytes ip = 1;
  required int32 port = 2;
}

message DataCmdRequest {
  enum TYPE {
    SYNC = 1;
  }
  required TYPE type = 1;

  // Join 
  message Sync {
    required Node node = 1;
    required int32 filenum = 2 [default = 0];
    required int32 offset = 3 [default = 0];
  }
  optional Sync sync = 2; 
}

message DataCmdResponse {
  enum TYPE {
    SYNC = 1;
  }
  required TYPE type = 1;

  message Status {
    required int32 status = 1;
    optional bytes msg = 2;
  }

  required Status status = 2;
}
